rpcserver.go

```diff
@@ -6777,7 +6779,35 @@ func (r *rpcServer) ForwardingHistory(ctx context.Context,
 	}
 	timeSlice, err := r.server.miscDB.ForwardingLog().Query(eventQuery)
 	if err != nil {
-		return nil, fmt.Errorf("unable to query forwarding log: %v", err)
+		return nil, fmt.Errorf("unable to query forwarding log: %v",
+			err)
+	}
+
+	// Helper function to extract a peer's node alias given its SCID.
+	getRemoteAlias := func(chanId lnwire.ShortChannelID) string {
```

Comment:
I think it is very likely that we will have multiple forwards for the same channel. So how about adding a lookup map here too so that if we have previously called `GetChanInfo` for a chanID, then we can get the alias from the map instead of doing the call again

Reply:
That's a very good idea to cache the channel/alias here! Just to confirm are you thinking of having the map in the rpcserver struct?
https://github.com/lightningnetwork/lnd/blob/master/rpcserver.go#L588

And in the case of a shutdown/restart I don't think we have to write the map to the db as it can get refilled when lnd is up again. Do you agree?

Reply:
Oh I see, so this caching would happen per `ForwardingHistory` call whereas if we had it in the `rpcServer` struct we could cache across different `ForwardingHistory` calls but it would cost more memory.

So I'll go for your suggestion for now.

Reply:
thx @ellemouton, I've added your suggestion. The itest is covering the for the repeated lookup.